elastic-query-generator/README.md

```diff
@@ -0,0 +1,21 @@
+# elastic-query-generator
+
+We can use LLMs to interact with Elasticsearch analytics databases in natural language.
+
+This chain builds search queries via the Elasticsearch DSL API (filters and aggregations).
+
+The Elasticsearch client must have permissions for index listing, mapping description and search queries.
+
+
+
+## Setup
+
+See [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html) for instructions on how to run Elasticsearch locally.
+
+In addition to the instructions there, you will want to run
+
+> export ELASTIC_SEARCH_SERVER="https://elastic:pass@localhost:9200"
+
+Note that if you already have an Elastic cluster running you can just replace the code to call out to that.
```

Comment:
```suggestion
## Installing Elasticsearch

There are a number of ways to run Elasticsearch.

### Elastic Cloud

Create a free trial account on [Elastic Cloud](https://cloud.elastic.co/registration?utm_source=langchain&utm_content=langserve).

With a deployment, update the connection string.

Password and connection (elasticsearch url) can be found on the deployment console.  

```bash
> export ELASTIC_SEARCH_SERVER="https://elastic:<password>@<es-url>"
```

### Docker

For local development, you can use Docker:

```bash
docker run -p 9200:9200 \
  -e "discovery.type=single-node" \
  -e "xpack.security.enabled=false" \
  -e "xpack.security.http.ssl.enabled=false" \
  -e "xpack.license.self_generated.type=trial" \
  docker.elastic.co/elasticsearch/elasticsearch:8.10.0
```

This will run Elasticsearch on port 9200. You can then check that it is running by visiting [http://localhost:9200](http://localhost:9200).

In addition to the instructions there, you will want to run

```bash
> export ELASTIC_SEARCH_SERVER="http://localhost:9200"
```

Reply:
Hey change suggestion here just to make it a little more simple to run by switching off security and SSL. This is fine for local "try it out mode" and not to complicate things with custom ssl certs and password.

Also added elastic cloud connection notes.